### DISCUSSION TITLE
Import from anonymous

### DISCUSSION THREAD

**Discussion by agelosc** [Autodesk/maya-usd#1069] [Q&A] [answered]
mayaUSDImport fails to import an anonymous layer. Is this a bug or limitation?

For example the following code fails.
```
from pxr import Usd, Sdf
layer = Sdf.Layer.CreateAnonymous()
stage = Usd.Stage.Open(layer)
stage.DefinePrim('/Foo', 'Xform')
cmds.mayaUSDImport(layer.identifier, primPath='/')
```
TypeError: anon:.. is invalid

**Comment by kxl-adsk** [comment, marked as answer]
When I look at the job code, I'm having the impression that what you are trying to do might be possible. But clearly, the command expects you to pass the file path. Your above example fails during argument parsing, but when you fix it that and call `cmds.mayaUSDImport(f=layer.identifier, primPath='/')` you will get an error from this validation - https://github.com/Autodesk/maya-usd/blob/6a9f673831d504550d4badf083cb024de87bcb73/lib/mayaUsd/commands/baseImportCommand.cpp#L128

So to answer your question directly, this looks like a limitation.

**Comment by agelosc** [comment]
mayaUSDExport works as expected when an anonymous layer is passed to the filePath argument so I assumed and hoped import would be the same. I don't understand why this limitation exists but thank you for having a look.

**Comment by kxl-adsk** [comment]
Thank you for mentioning export. I wasn't aware that we didn't have symmetry between the two. Then I consider this a bug (even if maybe confusing to pass identifier as the file path).

The fix is in review - https://github.com/Autodesk/maya-usd/pull/1086. Try it if you are compiling MayaUSD by yourself.

**Comment by agelosc** [comment]
Sorry I didn't mentioned it before, the export command with a layer identifier is used in [tests](https://github.com/Autodesk/maya-usd/blob/cd192e8af2cd86421eadd0a98fb2903f70d6e078/test/lib/usd/translators/testUsdExportOpenLayer.py#L78) so I assumed it's standard functionality.

Compiling is a bit more than I can chew but I'll wait for the next release. This is great news, thank you.
